### DISCUSSION TITLE
How to use the port forwarding function on the browser

### DISCUSSION THREAD

**Discussion by hello-zhanghao** [gitpod-io/openvscode-server#260] [General]
I want to use latex workshop in my openvscode server, but when I want to see the pdf in openvscode, it needs to request a port like localhost:49123 . In  vscode, there are a port forwarding function can forward requests to the server， for example localhost:49123 -> server:49123, so I can see the pdf in vscode. But in the openvscode, there are no port forwarding. 
So, there are any extentions or onthers can Implement this port forwarding in openvscoe?

**Comment by akosyakov** [comment]
There is no such extensions yet for OpenVSCode Server. You will need to use your hoisting infrastructure, i.e. port forwarding features of Docker, SSH or proxies. It depends how you deploy it. Please try to search in other discussions.

**Reply by hello-zhanghao** [reply]
In the gitpod Workspace, there are remote explorer can implement the port forwarding function in the browser. I want to know
How did you do that！extension or what？Could you add  this capability to the openvscoe? Thanks
![image](https://user-images.githubusercontent.com/38358269/146668368-04fdd99d-2342-493a-9488-509430f7ede1.png)

**Reply by akosyakov** [reply]
We use k8s to open such ports. If you want such features you should look into self hosting Gitpod.

Something like that can implemented with integration to other tools capable of port forwarding like docker and SSH servers, but it is not a priority right now.

**Comment by drupol** [comment]
I'm still looking for a solution for this, I just opened an issue #515 

I mean, if it's doable with coder/code-server, why isn't it with gitpod-io/openvscode-server ? What's preventing that feature to be implemented?

If we need to setup our own tunnel with SSH, how to know in advance which ports to map ?

**Comment by LinuxSuRen** [comment]
I am trying to find a way to listen to the port forwarding event. But I didn't see it. I can create a server to expose the port in Kubernetes as ingress, but don't know how to get the listening port. Do you have any ideas?

**Comment by yeldhomathew** [comment]
As per above discussion, We cannot have port forwarding feature in Openvscode browser version. Is my Understanding correct? Is there any hack to get port forwarding feature in Openvscode Browser version.

**Reply by drupol** [reply]
> As per above discussion, We cannot have port forwarding feature in Openvscode browser version. Is my Understanding correct? Is there any hack to get port forwarding feature in Openvscode Browser version.

Yes, none, use coder for that.
